### PR TITLE
LAS 3.0 : Rearrange reader branch: Fix find_sections_in_file with text stream

### PR DESCRIPTION
**Pull request by dcslagel**
This pull request is an alternative to https://github.com/kinverarity1/lasio/pull/357.

This request retains most of the readline() approach with changes to get the correct section positions.  It also replaces codecs.open() with the more current io.open.  However the other codecs encoding logic is still there.

It passes the test suite on Python 3.7  with 21 tests failing

Currently it fails to complete the test suite on Python 2.7 with the error: 
```UnicodeEncodeError: 'ascii' codec can't encode characters```
This can be worked around.

Is either approach more preferred than the other?

Thanks!

DC

### CODE REVIEW

**Comment by kinverarity1** [comment]
Thanks for both of these! I will have a closer look this weekend...

**Comment by dcslagel** [comment]
Great.  Thanks, I appreciate your taking the time to look these over.

File: lasio/reader.py [hunk 1]
```diff
@@ -270,27 +271,33 @@ def find_sections_in_file(file_obj):
...
 
     """
-    k = 0
+    file_pos = int(file_obj.tell())
     starts = []
     ends = []
-    for i, line in enumerate(file_obj):
+    line_no = 0
+    line = file_obj.readline()
+    # for i, line in enumerate(file_obj):
```
Comment (dcslagel) [inline lasio/reader.py:286]: Note: io.TextIOWrapper 'locks' file_obj.tell() if run in a for loop but doesn't in the while statement.  That is why the change from ```for``` to ```while```.

File: lasio/reader.py [hunk 2]
```diff
@@ -270,27 +271,33 @@ def find_sections_in_file(file_obj):
...
+    while line:
         sline = line.strip().strip("\n")
         if sline.startswith("~"):
-            starts.append((k, i, sline))
+            starts.append((file_pos, line_no, sline))
```
Comment (dcslagel) [inline lasio/reader.py:290]: Changing ```k``` and ```i``` to descriptive names ```file_pos``` and ```line_no``` just helped me keep the logic in my head.  Let me know if ```k``` and ```i``` are preferred.

**Comment by dcslagel** [comment]
Yes, I agree with you, #358 is clearer and will be more manageable to maintain.  Thank you for making the time to review these!

Regarding the issue in the Python 2.7 tests `UnicodeEncodeError: 'ascii' codec can't encode characters`,  what are the Lasio plans for Python 2.7 support?  How long will Lasio continue support for 2.7? 

It looks like the Python Foundation ended support on January 1, this year.
https://www.python.org/doc/sunset-python-2/

Thanks,
DC

**Comment by kinverarity1** [comment]
We can drop support for 2.7 as soon as we have to... It's time 😎
